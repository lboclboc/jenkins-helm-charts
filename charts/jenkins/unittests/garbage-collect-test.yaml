suite: Controller Admin Credentials
release:
  name: my-release
  namespace: my-namespace
templates:
  - jcasc-config.yaml
tests:
  - it: one cloud
    set:
        agent.garbageCollection.enabled: true
    asserts:
      - isKind:
          of: ConfigMap
      - hasDocuments:
         count: 1
      - isNotEmpty:
          path: data["jcasc-default-config.yaml"]
      - matchRegex:
          path: metadata.labels["helm.sh/chart"]
          pattern: ^jenkins-
      - matchSnapshot:
          path: data["jcasc-default-config.yaml"]

  - it: second cloud
    set:
        agent.garbageCollection.enabled: true
        additionalClouds.second.agent.containerCap: 5
    asserts:
      - isKind:
          of: ConfigMap
      - hasDocuments:
         count: 1
      - isNotEmpty:
          path: data["jcasc-default-config.yaml"]
      - matchRegex:
          path: metadata.labels["helm.sh/chart"]
          pattern: ^jenkins-
      - matchSnapshot:
          path: data["jcasc-default-config.yaml"]

