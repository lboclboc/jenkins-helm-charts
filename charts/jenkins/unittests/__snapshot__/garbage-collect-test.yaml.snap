one cloud:
  1: |
    |-
      jenkins:
        authorizationStrategy:
          loggedInUsersCanDoAnything:
            allowAnonymousRead: false
        securityRealm:
          local:
            allowsSignup: false
            enableCaptcha: false
            users:
            - id: "${chart-admin-username}"
              name: "Jenkins Admin"
              password: "${chart-admin-password}"
        disableRememberMe: false
        mode: NORMAL
        numExecutors: 0
        labelString: ""
        projectNamingStrategy: "standard"
        markupFormatter:
          plainText
        clouds:
        - kubernetes:
            containerCapStr: "10"
            garbageCollection:
              timeout: "300"
            defaultsProviderTemplate: ""
            connectTimeout: "5"
            readTimeout: "15"
            jenkinsUrl: "http://my-release-jenkins.my-namespace.svc.cluster.local:8080"
            jenkinsTunnel: "my-release-jenkins-agent.my-namespace.svc.cluster.local:50000"
            skipTlsVerify: false
            usageRestricted: false
            maxRequestsPerHostStr: "32"
            retentionTimeout: "5"
            waitForPodSec: "600"
            name: "kubernetes"
            namespace: "my-namespace"
            restrictedPssSecurityContext: false
            serverUrl: "https://kubernetes.default"
            credentialsId: ""
            podLabels:
            - key: "jenkins/my-release-jenkins-agent"
              value: "true"
            templates:
              - name: "default"
                namespace: "my-namespace"
                id: 56ff30d0e9a4d255e93a29a1a9cec83f121c505e0d5ced46cccf0589f928c0ad
                containers:
                - name: "jnlp"
                  alwaysPullImage: false
                  args: "^${computer.jnlpmac} ^${computer.name}"
                  envVars:
                    - envVar:
                        key: "JENKINS_URL"
                        value: "http://my-release-jenkins.my-namespace.svc.cluster.local:8080/"
                  image: "jenkins/inbound-agent:3301.v4363ddcca_4e7-3"
                  privileged: "false"
                  resourceLimitCpu: 512m
                  resourceLimitMemory: 512Mi
                  resourceRequestCpu: 512m
                  resourceRequestMemory: 512Mi
                  ttyEnabled: false
                  workingDir: /home/jenkins/agent
                idleMinutes: 0
                instanceCap: 2147483647
                label: "my-release-jenkins-agent "
                nodeUsageMode: "NORMAL"
                podRetention: Never
                showRawYaml: true
                serviceAccount: "default"
                slaveConnectTimeoutStr: "100"
                yamlMergeStrategy: override
                inheritYamlMergeStrategy: false
        slaveAgentPort: 50000
        crumbIssuer:
          standard:
            excludeClientIPFromCrumb: true
      security:
        apiToken:
          creationOfLegacyTokenEnabled: false
          tokenGenerationOnCreationEnabled: false
          usageStatisticsEnabled: true
      unclassified:
        location:
          url: http://my-release-jenkins:8080
second cloud:
  1: |
    |-
      jenkins:
        authorizationStrategy:
          loggedInUsersCanDoAnything:
            allowAnonymousRead: false
        securityRealm:
          local:
            allowsSignup: false
            enableCaptcha: false
            users:
            - id: "${chart-admin-username}"
              name: "Jenkins Admin"
              password: "${chart-admin-password}"
        disableRememberMe: false
        mode: NORMAL
        numExecutors: 0
        labelString: ""
        projectNamingStrategy: "standard"
        markupFormatter:
          plainText
        clouds:
        - kubernetes:
            containerCapStr: "10"
            garbageCollection:
              timeout: "300"
            defaultsProviderTemplate: ""
            connectTimeout: "5"
            readTimeout: "15"
            jenkinsUrl: "http://my-release-jenkins.my-namespace.svc.cluster.local:8080"
            jenkinsTunnel: "my-release-jenkins-agent.my-namespace.svc.cluster.local:50000"
            skipTlsVerify: false
            usageRestricted: false
            maxRequestsPerHostStr: "32"
            retentionTimeout: "5"
            waitForPodSec: "600"
            name: "kubernetes"
            namespace: "my-namespace"
            restrictedPssSecurityContext: false
            serverUrl: "https://kubernetes.default"
            credentialsId: ""
            podLabels:
            - key: "jenkins/my-release-jenkins-agent"
              value: "true"
            templates:
              - name: "default"
                namespace: "my-namespace"
                id: 56ff30d0e9a4d255e93a29a1a9cec83f121c505e0d5ced46cccf0589f928c0ad
                containers:
                - name: "jnlp"
                  alwaysPullImage: false
                  args: "^${computer.jnlpmac} ^${computer.name}"
                  envVars:
                    - envVar:
                        key: "JENKINS_URL"
                        value: "http://my-release-jenkins.my-namespace.svc.cluster.local:8080/"
                  image: "jenkins/inbound-agent:3301.v4363ddcca_4e7-3"
                  privileged: "false"
                  resourceLimitCpu: 512m
                  resourceLimitMemory: 512Mi
                  resourceRequestCpu: 512m
                  resourceRequestMemory: 512Mi
                  ttyEnabled: false
                  workingDir: /home/jenkins/agent
                idleMinutes: 0
                instanceCap: 2147483647
                label: "my-release-jenkins-agent "
                nodeUsageMode: "NORMAL"
                podRetention: Never
                showRawYaml: true
                serviceAccount: "default"
                slaveConnectTimeoutStr: "100"
                yamlMergeStrategy: override
                inheritYamlMergeStrategy: false
        - kubernetes:
            containerCapStr: "5"
            garbageCollection:
              timeout: "300"
            defaultsProviderTemplate: ""
            connectTimeout: "5"
            readTimeout: "15"
            jenkinsUrl: "http://my-release-jenkins.my-namespace.svc.cluster.local:8080"
            jenkinsTunnel: "my-release-jenkins-agent.my-namespace.svc.cluster.local:50000"
            skipTlsVerify: false
            usageRestricted: false
            maxRequestsPerHostStr: "32"
            retentionTimeout: "5"
            waitForPodSec: "600"
            name: "second"
            namespace: "my-namespace"
            restrictedPssSecurityContext: false
            serverUrl: "https://kubernetes.default"
            credentialsId: ""
            podLabels:
            - key: "jenkins/my-release-jenkins-agent"
              value: "true"
            templates:
              - name: "default"
                namespace: "my-namespace"
                id: 88230fb429b3839e91ce04012b6a28428772ea76d52885cc2e87bdf19035457a
                containers:
                - name: "jnlp"
                  alwaysPullImage: false
                  args: "^${computer.jnlpmac} ^${computer.name}"
                  envVars:
                    - envVar:
                        key: "JENKINS_URL"
                        value: "http://my-release-jenkins.my-namespace.svc.cluster.local:8080/"
                  image: "jenkins/inbound-agent:3301.v4363ddcca_4e7-3"
                  privileged: "false"
                  resourceLimitCpu: 512m
                  resourceLimitMemory: 512Mi
                  resourceRequestCpu: 512m
                  resourceRequestMemory: 512Mi
                  ttyEnabled: false
                  workingDir: /home/jenkins/agent
                idleMinutes: 0
                instanceCap: 2147483647
                label: "my-release-jenkins-agent "
                nodeUsageMode: "NORMAL"
                podRetention: Never
                showRawYaml: true
                serviceAccount: "default"
                slaveConnectTimeoutStr: "100"
                yamlMergeStrategy: override
                inheritYamlMergeStrategy: false
        slaveAgentPort: 50000
        crumbIssuer:
          standard:
            excludeClientIPFromCrumb: true
      security:
        apiToken:
          creationOfLegacyTokenEnabled: false
          tokenGenerationOnCreationEnabled: false
          usageStatisticsEnabled: true
      unclassified:
        location:
          url: http://my-release-jenkins:8080
